packages/usa-file-input/src/index.js

```diff
@@ -393,37 +409,25 @@ const handleChange = (e, fileInputEl, instructions, dropTarget) => {
     // Not all files will be able to generate previews. In case this happens, we provide several types "generic previews" based on the file extension.
     reader.onloadend = function createFilePreview() {
       const previewImage = document.getElementById(imageId);
-      if (fileName.indexOf(".pdf") > 0) {
-        previewImage.setAttribute(
-          "onerror",
-          `this.onerror=null;this.src="${SPACER_GIF}"; this.classList.add("${PDF_PREVIEW_CLASS}")`,
-        );
-      } else if (
-        fileName.indexOf(".doc") > 0 ||
-        fileName.indexOf(".pages") > 0
-      ) {
-        previewImage.setAttribute(
-          "onerror",
-          `this.onerror=null;this.src="${SPACER_GIF}"; this.classList.add("${WORD_PREVIEW_CLASS}")`,
-        );
-      } else if (
-        fileName.indexOf(".xls") > 0 ||
-        fileName.indexOf(".numbers") > 0
-      ) {
-        previewImage.setAttribute(
-          "onerror",
-          `this.onerror=null;this.src="${SPACER_GIF}"; this.classList.add("${EXCEL_PREVIEW_CLASS}")`,
-        );
-      } else if (fileName.indexOf(".mov") > 0 || fileName.indexOf(".mp4") > 0) {
-        previewImage.setAttribute(
-          "onerror",
-          `this.onerror=null;this.src="${SPACER_GIF}"; this.classList.add("${VIDEO_PREVIEW_CLASS}")`,
-        );
-      } else {
-        previewImage.setAttribute(
-          "onerror",
-          `this.onerror=null;this.src="${SPACER_GIF}"; this.classList.add("${GENERIC_PREVIEW_CLASS}")`,
-        );
+      const fileExtension = fileName.split(".").pop();
+      switch (fileExtension) {
+        case "pdf":
+          setPreviewFallback(previewImage, PDF_PREVIEW_CLASS);
+          break;
+        case "doc":
+        case "pages":
+          setPreviewFallback(previewImage, WORD_PREVIEW_CLASS);
+          break;
+        case "xls":
+        case "numbers":
+          setPreviewFallback(previewImage, EXCEL_PREVIEW_CLASS);
+          break;
+        case "mov":
+        case "mp4":
+          setPreviewFallback(previewImage, VIDEO_PREVIEW_CLASS);
+          break;
+        default:
+          setPreviewFallback(previewImage, GENERIC_PREVIEW_CLASS);
```

Comment:
@jeffpw-goog are you still around to help with this issue?

Switch cases aren't common in USWDS code. I was wondering if you'd be open to refactoring back to an `if..else` statement to match USWDS code patterns!

Reply:
Done